### DISCUSSION TITLE
When will PixiJS v7.0.0 be finally release?

### DISCUSSION THREAD

**Discussion by SuperSodaSea** [pixijs/pixijs#8739] [General]
cc @bigtimebuddy

**Comment by bigtimebuddy** [comment]
Soon. Maybe this week? 

There are a few immediate goals for getting v7 released:

* Identify any blocker issues (this is on-going, folks upgrading now to v7 will help us a lot!)
* Improvements to the migration guide, if any
* Upgrade high-value plugins (spine, layers, etc)

To the last point: plugins can be the thing that holds folks back so getting those things upgraded ASAP is important.

**Reply by Cristy94** [reply]
Are there any blog posts/release notes regarding the changes introduced in v7?

**Reply by bigtimebuddy** [reply]
https://github.com/pixijs/pixijs/wiki/v7-Migration-Guide is probably the best overview of v7.

**Reply by Cristy94** [reply]
Thanks! So it's mostly an internal code change. It would be nice to also have some basic improvements/regressions benchmarks listed (performance, package size, loading times, etc.).

**Comment by bigtimebuddy** [comment]
Okay v7 is pretty much sorted at this point. Any last minute requests for must-have breaking changes? (ideally something easy, like default changes, ergonomics, etc)

Looking at you @dev7355608 @SuperSodaSea @ivanpopelyshev @miltoncandelero

**Reply by miltoncandelero** [reply]
One thing I don't remember what ended up being: Resolution inside render is now read only? How do we handle changes in the DevicePixelRatio if we can't change the resolution of the renderer? was there a function added somewhere to change it? 🤔 

Sorry I've been away during the development of this version, a big project caught fire and I am on firemen duty 🥲

**Reply by bigtimebuddy** [reply]
I don't think we've solved this yet. In general, resolution needs some work and better coordination internally. We have 2 global settings (`settings.RESOLUTION`, `settings.FILTER_RESOLUTION`), Renderer's constructor resolution, and we have local display object resolution (e.g., Text) and Filter's resolution AND BaseTexture's resolution. 

I would love us to rethink resolution conceptually to 1) make it simpler to use and understand, 2) support more dynamic uses cases like browser zoom, 3) to allow for optimization/overriding when needed, and 4) provide better user-feedback when user does something like set an asset's resolution higher than the Renderer resolution. 

Maybe having an observable for resolution and then a local/manual resolution to override? I think this needs a proposal.

**Reply by miltoncandelero** [reply]
I haven't seen the old nor the new code for renderer, however the old one had a fully writable resolution field and that worked just fine 🤔 

It was a quick and dirty way to implement a "quality" slider to allow for uglier graphics and faster performance (as it would downscale the internal rendering resolution).

However, I agree we should have a more accessible and documented solution in place 👍 
Since I don't have the time right now to dive into the Renderer code and think up a elegant implementation, it will have to wait to future patches 😅

**Reply by bigtimebuddy** [reply]
There was a reason we made it read-only after construction, and I can't remember why. Maybe @GoodBoyDigital remembers? There still is a field, but it's `renderer._view.resolution`, for hacky solutions.

**Reply by miltoncandelero** [reply]
> `renderer._view.resolution`, for hacky solutions.

as long as we have a hacky way, I am fine with it 👍 
I will make an RFC in the fuuuuture about it

**Comment by bigtimebuddy** [comment]
v7.0.0 is published!
